File: lib/widgets/attendance_flow/take_care_sheet.dart
import 'package:cached_network_image/cached_network_image.dart';
import 'package:flutter/material.dart';
import 'package:flutter_screenutil/flutter_screenutil.dart';
import 'package:provider/provider.dart';
import 'package:snabbit_runner/providers/language_provider.dart';
import 'package:snabbit_runner/services/mixpanel_setup.dart';
import 'package:snabbit_runner/utils/tracking_events.dart';

/// Standalone "Take care!" bottom sheet shown after successful period leave.
///
/// Figma node 7455:41537 — white bg, 24 top / 40 bottom padding, 24 top radius,
/// heart illustration, title + caption, full-width primary button.
class TakeCareSheet extends StatelessWidget {
  const TakeCareSheet({super.key});

  // Figma colour tokens
  static const _gray800 = Color(0xFF1F2937);
  static const _gray500 = Color(0xFF6B7280);
  static const _gray900 = Color(0xFF111827);

  /// [source] identifies which flow opened this sheet, for analytics.
  /// Values: 'emergency_logout', 'false_attendance'.
  static Future<void> show(BuildContext context, {required String source}) {
    final props = <String, dynamic>{'source': source};
    MixpanelSetup.logEvent(TrackingEvents.takeCareBsLoad, props);
    return showModalBottomSheet<void>(
      context: context,
      isDismissible: true,
      enableDrag: true,
      backgroundColor: Colors.transparent,
      builder: (_) => const TakeCareSheet(),
    );
  }

  @override
  Widget build(BuildContext context) {
    final lang = context.watch<LanguageProvider>();

    final bottomInset = MediaQuery.viewPaddingOf(context).bottom;

    return Container(
      decoration: BoxDecoration(
        color: Colors.white,
        borderRadius: BorderRadius.only(
          topLeft: Radius.circular(24.r),
          topRight: Radius.circular(24.r),
        ),
      ),
      child: Padding(
        padding: EdgeInsets.only(
          left: 16.w,
          right: 16.w,
          top: 24.h,
          bottom: 40.h + bottomInset,
        ),
        child: Column(
          mainAxisSize: MainAxisSize.min,
          children: [
            // Heart illustration + title + subtitle — 16 between art and text, 4 between lines
            Column(
              mainAxisSize: MainAxisSize.min,
              children: [
                // Heart illustration — Figma: 100×100
                CachedNetworkImage(
                  imageUrl:
                      'https://assets-expert.snabbit.com/period_leave/take_care_heart.png',
                  width: 100.r,
                  height: 100.r,
                  fit: BoxFit.contain,
                  errorWidget: (_, __, ___) => Container(
                    width: 100.r,
                    height: 100.r,
                    decoration: const BoxDecoration(
                      shape: BoxShape.circle,
                      color: Color(0xFFFCE4EC),
                    ),
                    child: Icon(
                      Icons.favorite,
                      size: 50.sp,
                      color: Colors.redAccent,
                    ),
                  ),
                ),

                SizedBox(height: 16.h),

                // Title + subtitle — gap 4
                Column(
                  mainAxisSize: MainAxisSize.min,
                  children: [
                    // "Take care!" — Outfit Semibold 24sp, #1f2937
                    Text(
                      lang.getMessage('take_care', 'Take care!'),
                      textAlign: TextAlign.center,
                      style: TextStyle(
                        fontFamily: 'Outfit',
                        color: _gray800,
                        fontSize: 24.sp,
                        fontWeight: FontWeight.w600,
                        height: 32 / 24,
                      ),
                    ),
                    SizedBox(height: 4.h),
                    // "No red card applied" — Caption/12-Medium, #6b7280
                    Text(
                      lang.getMessage(
                        'no_red_card_applied',
                        'No red card applied',
                      ),
                      textAlign: TextAlign.center,
                      style: TextStyle(
                        fontFamily: 'Outfit',
                        color: _gray500,
                        fontSize: 12.sp,
                        fontWeight: FontWeight.w500,
                        height: 16 / 12,
                      ),
                    ),
                  ],
                ),
              ],
            ),

            SizedBox(height: 40.h),

            // Primary button — gray-900, 48h, radius 8, full width
            SizedBox(
              width: double.infinity,
              height: 48.h,
              child: ElevatedButton(
                onPressed: () => Navigator.of(context).pop(),
                style: ElevatedButton.styleFrom(
                  backgroundColor: _gray900,
                  shape: RoundedRectangleBorder(
                    borderRadius: BorderRadius.circular(8.r),
                  ),
                  elevation: 0,
                ),
                child: Text(
                  lang.getMessage('ok', 'Okay'),
                  style: TextStyle(
                    fontFamily: 'Outfit',
                    color: Colors.white,
                    fontSize: 16.sp,
                    fontWeight: FontWeight.w500,
                    height: 24 / 16,
                  ),
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }
}

File: shared/src/commonMain/kotlin/com/snabbit/runner/shared/features/shift/presentation/emergencylogout/TakeCareSheet.kt
package com.snabbit.runner.shared.features.shift.presentation.emergencylogout
import com.snabbit.design.atoms.SnabbitImage
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import com.snabbit.design.atoms.SnabbitButton
import com.snabbit.design.atoms.SnabbitButtonSize
import com.snabbit.design.atoms.SnabbitButtonStyle
import com.snabbit.design.atoms.SnabbitText
import com.snabbit.design.atoms.SnabbitTextVariant
import com.snabbit.runner.shared.resources.Res
import com.snabbit.runner.shared.resources.take_care_heart
import org.jetbrains.compose.resources.painterResource
import com.snabbit.design.theme.SnabbitColorsLight
import com.snabbit.design.theme.SnabbitTheme

/**
 * "Take care!" sheet — Figma 7455:41537 / Dart `take_care_sheet.dart`. Shown
 * after a successful period-leave emergency logout.
 *
 * Layout (top → bottom): 100dp heart illustration, "Take care!" title,
 * "No red card applied" subtitle, full-width "Okay" CTA. Heart asset is
 * bundled (`take_care_heart.png` from Dart's `assets-expert.snabbit.com`).
 */
@Composable
fun TakeCareSheet(
    strings: com.snabbit.runner.shared.features.shift.presentation.emergencylogout.EmergencyLogoutStrings,
    onDone: () -> Unit,
    modifier: Modifier = Modifier,
) {
    Column(
        modifier = modifier
            .fillMaxWidth()
            .padding(horizontal = SnabbitTheme.spacing.componentPaddingMd, vertical = SnabbitTheme.spacing.componentPaddingSm),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(SnabbitTheme.spacing.componentGapLg),
    ) {
        SnabbitImage(
            painter = painterResource(Res.drawable.take_care_heart),
            contentDescription = null,
            modifier = Modifier.size(100.dp),
        )
        Column(
            modifier = Modifier.fillMaxWidth(),
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.spacedBy(SnabbitTheme.spacing.componentGapXs),
        ) {
            // "Take care!" — gray-800, 24sp semibold
            SnabbitText(
                text = strings.takeCareTitle,
                variant = SnabbitTextVariant.Heading2,
                fontWeight = FontWeight.SemiBold,
                color = SnabbitColorsLight.gray800,
                textAlign = TextAlign.Center,
            )
            // "No red card applied" — gray-500, 12sp medium
            SnabbitText(
                text = strings.takeCareBody,
                variant = SnabbitTextVariant.Caption,
                color = SnabbitTheme.colors.textSecondary,
                textAlign = TextAlign.Center,
            )
        }
        SnabbitButton(
            text = strings.takeCareCta,
            onClick = onDone,
            style = SnabbitButtonStyle.Primary,
            size = SnabbitButtonSize.L,
            fullWidth = true,
            modifier = Modifier.fillMaxWidth(),
        )
    }
}
